frontend/src/features/SideWindow/MissionForm/MainForm/FormikMultiControlUnitPicker/ControlUnitSelect.tsx

```diff
@@ -0,0 +1,233 @@
+import type { MissionFormValues } from '../types'
+import type { Promisable } from 'type-fest'
+
+export type ControlUnitSelectProps = {
+  administrationsAsOptions: Option[]
+  controlUnits: ControlUnit[] | undefined
+  defaultValue: MissionFormValues['controlUnits'][0]
+  index: number
+  onChange: (index: number, nextControlUnit: MissionFormValues['controlUnits'][0]) => Promisable<void>
+  onDelete: (index: number) => Promisable<void>
+}
+export function ControlUnitSelect({
+  administrationsAsOptions,
+  controlUnits,
+  defaultValue,
+  index,
+  onChange,
+  onDelete
+}: ControlUnitSelectProps) {
+  const controlledValueRef = useRef(defaultValue)
+  const { newWindowContainerRef } = useNewWindow()
+
+  const [selectedControlUnit, setSelectedControlUnit] = useState<ControlUnit | undefined>(undefined)
+
+  const { forceUpdate } = useForceUpdate()
+
+  const resourcesAsOptions = useMemo((): Option<number>[] => {
+    if (!selectedControlUnit) {
+      return []
+    }
+
+    return mapControlUnitToResourcesAsOptions(selectedControlUnit)
+  }, [selectedControlUnit])
+
+  const controlledValueResourceIds = useMemo(
+    () => (controlledValueRef.current.resources ? mapToProp(controlledValueRef.current.resources, 'id') : []),
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [controlledValueRef.current]
+  )
+
+  const unitsAsOptions = useMemo(
+    (): Option[] => {
+      if (!controlUnits || !controlledValueRef.current.administration) {
+        return []
+      }
+
+      const selectedAdministrationControlUnits = controlUnits.filter(
+        ({ administration }) => administration === controlledValueRef.current.administration
+      )
+
+      return mapControlUnitsToUniqueSortedNamesAsOptions(selectedAdministrationControlUnits)
+    },
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [controlledValueRef.current, controlUnits]
+  )
+
+  const handleAdministrationChange = useCallback(
+    (nextAdministration: string | undefined) => {
+      const nextControlUnit: MissionFormValues['controlUnits'][0] = {
+        ...INITIAL_MISSION_CONTROL_UNIT,
+        administration: nextAdministration
+      }
+
+      controlledValueRef.current = nextControlUnit
+      forceUpdate()
+
+      onChange(index, nextControlUnit)
+    },
+    [forceUpdate, index, onChange]
+  )
+
+  const handleNameChange = useCallback(
+    (nextName: string | undefined) => {
+      if (!controlUnits || !controlledValueRef.current.administration) {
+        return
+      }
+
+      const nextSelectedControlUnit = nextName
+        ? findControlUnitByAdministrationAndName(controlUnits, controlledValueRef.current.administration, nextName)
+        : undefined
+
+      const nextControlUnit: MissionFormValues['controlUnits'][0] = {
+        ...INITIAL_MISSION_CONTROL_UNIT,
+        ...pick(['administration'])(controlledValueRef.current),
+        id: nextSelectedControlUnit ? nextSelectedControlUnit.id : undefined,
```

Comment:
If the `controlUnit` has no `id`, we can't save it, so `undefined` is not allowed. Or maybe I missed something ?

Reply:
When you create a new ControlUnit, it doesn't have an ID, it's the database that generates the INSERT ID no?

Reply:
We are not allowed to create a new ControlUnit, we only select onr from the available list

Reply:
Sorry I replied too fast, I thought you were talking about the .

The undefined here is to handle the unselection. If name selected is cleaned, the the id must go back to `undefined`.